packages/eslint-plugin/src/rules/prefer-at.ts

```diff
@@ -0,0 +1,177 @@
+    }
+
+    const supporterObjects: Array<SupportedObject> = [
+      checkObjectName('Array'),
+      checkObjectName('Int8Array'),
+      checkObjectName('Uint8Array'),
+      checkObjectName('Uint8ClampedArray'),
+      checkObjectName('Int16Array'),
+      checkObjectName('Uint16Array'),
+      checkObjectName('Int32Array'),
+      checkObjectName('Float32Array'),
+      checkObjectName('Uint32Array'),
+      checkObjectName('Float64Array'),
+      checkObjectName('BigInt64Array'),
+      checkObjectName('BigUint64Array'),
+      // eslint-disable-next-line @typescript-eslint/internal/prefer-ast-types-enum
+      checkObjectName('String'),
+      checkObjectType(ts.TypeFlags.String),
+    ];
+
+    function isSupportedObject(type: ts.Type): boolean {
+      return supporterObjects.some(check => check(type));
+    }
+
+    function isExpectedObject(
+      node: TSESTree.Node,
+    ): node is TSESTree.MemberExpression {
+      if (
+        node.type !== AST_NODE_TYPES.MemberExpression ||
+        (options.ignoreFunctions && hasCallExpression(node))
+      ) {
+        return false;
+      }
+      const type = getTypeAtLocation(node.object);
+      if (!isSupportedObject(type)) {
+        return false;
+      }
+      const atMember = type.getProperty('at');
+      return Boolean(atMember);
+    }
+
+    function isExpectedExpressionLeft(
+      node: TSESTree.BinaryExpression,
+    ): boolean {
+      if (!isExpectedObject(node.left) || getName(node.left) !== 'length') {
+        return false;
+      }
+      const type = getTypeAtLocation(node.left);
+      return tsutils.isTypeFlagSet(type, ts.TypeFlags.NumberLike);
+    }
+
+    function isExpectedExpressionRight(
+      node: TSESTree.BinaryExpression,
+    ): boolean {
+      const type = getTypeAtLocation(node.right);
+      return tsutils.isTypeFlagSet(type, ts.TypeFlags.NumberLike);
+    }
+
+    function isExpectedExpression<T extends TSESTree.BinaryExpression>(
+      node: T,
+    ): node is T & { left: TSESTree.MemberExpression } {
+      return isExpectedExpressionRight(node) && isExpectedExpressionLeft(node);
+    }
+
+    return {
+      'MemberExpression[property.type="BinaryExpression"][property.operator="-"]'(
+        node: TSESTree.MemberExpressionComputedName & {
+          property: TSESTree.BinaryExpression & { operator: '-' };
+        },
+      ): void {
+        if (!isExpectedExpression(node.property)) {
+          return;
+        }
+        const objectName = getFullName(node.object);
+        const memberName = getFullName(node.property.left.object);
+        const rightName = getFullName(node.property.right);
+        if (objectName !== memberName) {
+          return;
+        }
+        context.report({
+          messageId: 'preferAt',
+          data: {
+            name: objectName,
+          },
+          node,
+          fix: fixer =>
+            fixer.replaceText(node, `${objectName}.at(-${rightName})`),
```

Comment:
[Question] Similar lint rules such as `prefer-regexp-exec` and `prefer-string-starts-ends-with` use a util called `getWrappingFixer`. Have you considered using it here? There are some edge cases in their tests I think we'd want for this one.